### ISSUE TITLE
FEATURE - Medium Widgets, View Maintenance Events

### ISSUE THREAD

**Issue by mikaelacaron** [mikaelacaron/Basic-Car-Maintenance#50]
## Describe the feature you want the app to have that it currently doesn't
* Create a medium widget that allows you to view the total number of maintenance events for a specific vehicle
Be able to configure which vehicle you want to view for this widget

## Describe alternatives you've considered
N/A
## Additional context
Only make the medium widget (I'll create new issues for adding different widget sizes later

**Comment by mikaelacaron** [comment]
@harryheres wanted to work on this, can you comment on this and I'll assign it to you

**Comment by harryheres** [comment]
@mikaelacaron Yes, I’m done with the #7 for now, so I can work on this :)

**Comment by mikaelacaron** [comment]
Notice on `dev` I created the Widget extension now, to hopefully cut down on merge conflicts

**Comment by harryheres** [comment]
Oh okay, thank you for the update!

**Comment by mikaelacaron** [comment]
Sorry! This task isn't ready to be worked on right now, because I haven't connected the models together, feel free to choose another issue and I'll leave you as an assignee and let you know when you can come back to work on this issue

**Comment by harryheres** [comment]
Okay, will stop working for the time being. Will be happy to continue once you've done the necessary features. Also, what models in particular are you referring to? Might be able to help out?

**Comment by mikaelacaron** [comment]
`MaintenanceEvent` and `Vehhicle` aren't connected, I would, but it might require how I am passing the models around, also I still need to change some Firebase rules

**Comment by harryheres** [comment]
@mikaelacaron How are we doing with this?

**Comment by mikaelacaron** [comment]
Once #141 is done, you're ready! Thanks for your patience!
I was going to do it myself, but kinda got caught up, so I created an issue for someone else to work on this

**Comment by harryheres** [comment]
> Once #141 is done, you're ready! Thanks for your patience! I was going to do it myself, but kinda got caught up, so I created an issue for someone else to work on this

Awesome, will be looking forward to it!

**Comment by mikaelacaron** [comment]
This can be worked on now! Thanks so much for the patience

**Comment by harryheres** [comment]
Awesome, will get to it ASAP!

**Comment by HarryHeres** [comment]
@mikaelacaron When creating the widget, it remains blank in the simulator 
<img width="367" alt="image" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/74781187/d264a423-6c17-42dd-9623-2b6ba3ab6cf4">

I suppose this should not happen? This is just the sample widget, just added the medium size. Or am I missing something?

**Comment by harryheres** [comment]
@mikaelacaron Do you want me to create new separate class for the medium widget or just modify the sample `BasicCarMaintenanceWidget` source file? 

Also, the widgets as provided at the moment, appear blank. I checked the code and it seems to me that it should be filled with couple of Simple Entries, each an hour apart (or a single entry as a placeholder). Or is there something I'm missing? 

<img width="396" alt="image" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/74781187/2d9ae23b-db45-4fe6-b7e1-69dfa7cdf24e">

UPDATE: When migrated to StaticConfiguration and TimelineProvider, the widgets show their contents. Seems like a bug inside the WidgetKit. Any ideas if there's a solution to this?

**Comment by mikaelacaron** [comment]
> UPDATE: When migrated to StaticConfiguration and TimelineProvider, the widgets show their contents. Seems like a bug inside the WidgetKit. Any ideas if there's a solution to this?


I'm not sure, it worked by default with the current code on my machine.

And yes go ahead and make a new class for the size

**Comment by harryheres** [comment]
Have you tested in on a real device or in a simulator?

**Comment by mikaelacaron** [comment]
Both, for the default widget, they work for me

**Comment by harryheres** [comment]
Hmm, okay, will try to inspect a bit more then.

**Comment by mikaelacaron** [comment]
Unfortunately I think I'm going to need to put this on hold again, because it's not going to filter properly, by vehicle, the way it's being saved right now

I thought it was, but looking at it closer I don't think it is

**Comment by harryheres** [comment]
Okay, let me know when it gets resolved :)

**Comment by mikaelacaron** [comment]
Sorry 😅 I just noticed it last night, will post more soon

**Comment by harryheres** [comment]
> Sorry 😅 I just noticed it last night, will post more soon

No worries, take your time :)

**Comment by mikaelacaron** [comment]
Thanks for your patience! I'm going to hold off on all the widget tasks for right now because I'm going to change how the vehicles and maintenance events are connected. But because it's so close to the end of Hacktoberfest I don't want to break everything for everyone, so I'll open this issue again after Hacktoberfest and you're welcome to contribute to it for fun if you'd like

**Comment by mikaelacaron** [comment]
Opening this back up for Hacktoberfest 2024!

**Comment by aking618** [comment]
@mikaelacaron Can I work on this? It'll be fun trying out working with widgets.

**Comment by mikaelacaron** [comment]
@aking618 you got this! 🔥

**Comment by aking618** [comment]
Progress Update:

I am able to load a user's vehicles as options for the widget and fetch maintenance events for them. Currently, I am hardcoding the user ID until I share the auth state with the widget.

<img width="369" alt="image" src="https://github.com/user-attachments/assets/ff73945d-3908-407d-a044-4e9cd2a0c7fa">
<br>
<img height="500" src="https://github.com/user-attachments/assets/d45beb6c-87b2-4501-a5da-4c874cd32d48">

### Next Steps
- Use the proper userID
- Finalize the layout

**Comment by mikaelacaron** [comment]
@aking618  great! make sure it's flexible so that the others who are working on the other widgets can also use it
